### DISCUSSION TITLE
Some straight Answers, probably..

### DISCUSSION THREAD

**Discussion by raiden076** [coqui-ai/TTS#2108] [General Q&A] [answered]
We needed to use voice cloning, on the fly, with minimal voice data (basically, exactly what YourTTS supposed to do with zero shot inference). Everything in English, just new speaker embedding.
This Will be used in production, and even any small amount data, **cannot** leave our environment.

[Coqui.ai](https://app.coqui.ai) has shown us quite interesting results. We were intrigued, and tried out the published multilingual YourTTS model, but quickly noticed, it is not even close to, what the webapp offers. No amount of tweaking here and there yielded any significant result.

We have gone through almost all the discussions, docs, whatever resources we could find, there is no proper training script, or any updated model/chkpt, or reason, why your hosted model is producing significantly better result than the published one, here. 
And surprisingly it seems, the authors are quietly ignoring everything regarding YourTTS, be it fine tuning or performance differences.

Though clearly stated [here](https://github.com/coqui-ai/TTS/discussions/1058#discussioncomment-1925808), this recipe is not the same as vits, another maintainer [mentions](https://github.com/coqui-ai/TTS/discussions/1249#discussioncomment-2197529) vits **is** the recipe to go for. We are aware that it uses the same backbone as vits, still technically, these are deliberate misinformation and misdirection.

What is exactly going on? From a discussion, mentioned above, the authors are trying to create a synthetic voice detection model, "to prevent misuse".
Quoting another user
> you're pursuing two mutually opposing goals, one of which will always gradually render the other obsolete and is therefore a waste of time.

Whatever your intentions are, regarding this great model, if someone clarifies that, It would be great.

And finally, Is there any way, to get the [Coqui.ai](https://app.coqui.ai) like results(new speaker embedding, from a small voice sample; and then inference, with comparable quality), locally. And if there is, how?

Forgive me for the long rant, But after naming it YourTTS/TTS for everyone, and behaving exactly opposite, wasting our hours upon hours time to find some solution, and then to fail eventually, It was not a great time spent with coqui-ai. 

referenced discussions: #1058 #1249

**Comment by king-dahmanus** [comment]
that model used in the coqui website is closed source. You'll have to collect a lot of good multi speaker data to try to replicate it. Also the sample rate was increased in the closed sourced model to around 22050 to sound higher quality. Yeah, don't know what's up with them though.

**Reply by raiden076** [reply]
It's just sad.. 😓
Just some clear instructions, like every other model, there wouldn't be any of these confusions..!!

**Comment by lexkoro** [comment, marked as answer]
All experiments as presented in the YourTTS paper were released here https://github.com/Edresson/YourTTS#reproducibility

The functionality currently available on the [Coqui.ai](https://app.coqui.ai/) website uses most likely an extended YourTTS model trained on higher quality data, which is noticeable in the audio quality produced. Since it is designed for commercial use, it is understandable why it is not open sourced.

Anyway, the original YourTTS code and models have been released and anyone is free to train a model themselves. :)
If you want to achieve good voice cloning (zero-shot), you need a large amount of voice data to train on, so that the model can generalize to a large number of speakers.

**Reply by raiden076** [reply]
Yes, I'm familiar with that repo, 
And regarding training.. 
That is fine, there are so many great multi-speaker datasets, to train with
But, before everything, after almost a year, they refuse to publish, the proper training script/recipe.
All we needed to be informed clearly, what will and won't be done, then there won't be these problems.

**Reply by lexkoro** [reply]
Refuse? 😅 The [reproducibility](https://github.com/Edresson/YourTTS#reproducibility) section describes the steps required to get started. 
But yeah, clearer documentation/instructions would probably lead to less confusion/ misconception.

**Reply by raiden076** [reply]
I really don't know how my post came out as,
The project and everything surrounding is great and, truly state of the art,

It's just the wild goose chase we had regarding this particular model, everything else, is so extensively documented, and there, the blog post about yourtts, hinted there are some elaborate recipe, (somewhere),
First we thought we missed somethings.. 

Just a bit better documentation, a lot of people won't have to go through this headache, i guess.
Many of them, for commercial or hobby purposes come to coqui for this exact purpose (open source voice cloning), and this is the exact place, where there's a huge gap in knowledge base!

**Reply by WeberJulian** [reply]
Thanks @lexkoro, that's our position. We released the model and config of the YourTTS paper where all the experiments, training procedures and data processing is explained. We extended the model and sell this new version. We don't have plans for open-sourcing the paid version we sell.
